spec/javascripts/components/admin_page_components/select_spec.js

```diff
@@ -0,0 +1,113 @@
+describe("Select", function() {
+  beforeEach(function() {
+    this.options = ['a', '1', '2', '3', '4', '5', 'all'];
+    this.value = 1
+    this.label = 'The label'
+
+    var component = React.createElement(Select, {
+      options: this.options,
+      value: this.value,
+      label: this.label
+    });
+    this.rendered = TestUtils.renderIntoDocument(component)
+  });
+
+  describe("constructor", function() {
+    it("it should create a uuid for itself", function() {
+      expect(this.rendered._uuid).not.toBe(undefined);
+    });
+  });
+
+  describe("generateSelectOptions", function() {
+    it("it should add the correct label", function() {
+      var result = this.rendered.generateSelectOptions(this.options);
+      var entry = result[0]
+      expect(entry.key).toEqual('def')
+      expect(entry.props.value).toEqual('def')
+      expect(entry.props.children).toEqual('Selecteer')
+    });
+
+    it("it should generate the select options using the options provided to it", function() {
+      var result = this.rendered.generateSelectOptions(this.options);
+
+      // +1 for the label
+      expect(result.length).toEqual(this.options.length + 1);
+      for (var i = 0, len = result.length; i < len; i++) {
+        if (i === 0) continue;
+        let entry = result[i];
+        expect(entry.key).toEqual(this.options[i - 1])
+        expect(entry.props.children).toEqual(this.options[i - 1])
+      }
+    });
+  });
+
+  describe("uuid", function() {
+    it("it should generate a random uuid", function() {
+      result = this.rendered.uuid()
+      result2 = this.rendered.uuid()
+      expect(result).not.toEqual(result2);
+    });
+
+    it("it should generate a uuid with the correct pattern", function() {
+      result = this.rendered.uuid()
+      expect(result.length).toEqual(36);
+      expect(result).toMatch(/^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-4[0-9a-fA-F]{3}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$/);
+    });
+  });
+
+  describe("redraw", function() {
+    it("it should call the material_select function on the select", function() {
+      // Don't know how to test this. Something like this?
+      //var select = ReactDOM.findDOMNode(this.rendered).children[0];
+      //spyOn(select, 'material_select');
+      //this.rendered.redraw()
+      //expect(select.material_select).toHaveBeenCalled()
+    });
```

Comment:
Ja dit voelt een beetje aan alsof je de internals van een functie aan het speccen bent, wat eigenlijk nooit hoeft. Volgens mij wil je gewoon testen dat als je de selectie aanpast, dattie dan de redraw functie aanroept. Wat die redraw functie zelf verder doet is niet zo super interessant (behalve dat je misschien ook nog zou kunnen testen dat als je een bepaald element aanklikt, dat het dan geselecteerd is na de redraw.